lib/rules/no-unused-vars.js

```diff
@@ -858,10 +861,357 @@ module.exports = {
+                                         * Gets fixes in variable declarations and function parameters
+                                         * @param {ASTNode} node parent node of identifier
+                                         * @returns {Object} fixer object
+                                         */
+                                        function fixVariables(node) {
+                                            if (node.parent.type === "VariableDeclarator") {
+                                                if (node.parent.parent.declarations.length === 1) {
+                                                    return fixer.removeRange(node.parent.parent.range);
+                                                }
+
+                                                if (sourceCode.getTokenBefore(node.parent).value === ",") {
+                                                    return fixer.removeRange([sourceCode.getTokenBefore(node.parent).range[0], node.parent.range[1]]);
+                                                }
+
+                                                return fixer.removeRange([node.parent.range[0], sourceCode.getTokenAfter(node.parent).range[1]]);
+                                            }
+
+                                            if (node.parent.type === "FunctionDeclaration" || node.parent.type === "FunctionExpression") {
+                                                if (node.parent.params.length === 1) {
+                                                    return fixer.removeRange(node.range);
+                                                }
+
+                                                if (sourceCode.getTokenBefore(node).value === "(" && sourceCode.getTokenAfter(node).value === ",") {
+                                                    return fixer.removeRange([node.range[0], sourceCode.getTokenAfter(node).range[1]]);
+                                                }
+
+                                                return fixer.removeRange([sourceCode.getTokenBefore(node).range[0], node.range[1]]);
+                                            }
+
+                                            return null;
+                                        }
+
+                                        /**
+                                         * Check if an array has a single element with null as other element.
+                                         * @param {ASTNode} node ArrayPattern node
+                                         * @returns {boolean} true if array has single element with other null elements
+                                         */
+                                        function hasSingleElement(node) {
+                                            return node.elements.filter(e => e !== null).length === 1;
+                                        }
+
+                                        if (parent.type === "VariableDeclarator") {
```

Comment:
Same comment about references throughout. We want to avoid a.b.c.d as much as possible.

Reply:
Done!